packages/tools/devtools/devtools-core/src/data-visualization/SharedTreeVisualizer.ts

```diff
@@ -302,3 +368,53 @@ export async function visualizeSharedTreeNodeBySchema(
 		}
 	}
 }
+
+/**
+ * Creates a visual representation of a SharedTree based on its schema.
+ * @param tree - The {@link VerboseTree} to visualize
+ * @param treeDefinitions - The schema that defines the structure and types of the tree
+ * @param allowedTypes - String representing the allowed types for this node
+ * @param isRequired - Whether this field is required in its parent object schema.
+ * Only meaningful for direct children of object nodes.
+ * Undefined for array/map elements since they are always required within their parent.
+ * @param visualizeChildData - Callback function to visualize child node data
+ * @returns A visual representation of the tree that includes schema information and node values
+ *
+ * @remarks
+ * This function handles both leaf nodes (primitive values, handles) and internal nodes (objects, maps, arrays).
+ * For leaf nodes, it creates a visual representation with the node's schema and value.
+ * For internal nodes, it recursively processes the node's fields using {@link visualizeInternalNodeBySchema}.
+ */
+export async function visualizeSharedTreeBySchema(
+	tree: VerboseTree,
+	treeDefinitions: ReadonlyMap<string, SimpleNodeSchema>,
+	allowedTypes: string,
+	isRequired: string | undefined,
```

Comment:
Could this be a `boolean` instead?

Reply:
This comment aligns with @alexvy86 's comment below. Will make the change.